### PR TITLE
Clean up the repository and enable easier installation

### LINKED ISSUE #14: MKIDGen3 waveform not importable without installing pynq
Culprit:

https://github.com/MazinLab/MKIDGen3/blame/94b2e6ba4060eac038e0425d13573a9ff4835249/mkidgen3/system_parameters.py#L45

and

https://github.com/MazinLab/MKIDGen3/blob/94b2e6ba4060eac038e0425d13573a9ff4835249/mkidgen3/drivers/__init__.py#L2

ifboard should probably be moved out of drivers if this is intended to be supported

### PR DESCRIPTION
**Pull request by ld-cd**
This PR is a big bundle of things, but each commit should be relatively self contained and can be cherry picked in if there are changes that need to be made to the PR overall.

Broadly this removes pieces of the repository that have gone stale and updates the python package to use the current standard python packaging metadata. This should make things like adding sphinx docs relatively straightforward and the correct dependency listing makes the client code pip installable.

The board side code is also pip installable with the exception of `xrfdc` and `xrfclk` packages but will check if they are installed, we may want to vendor these since we patch both of them anyways.

Fixes #14

### CODE REVIEW

File: mkidgen3/drivers/ifboard.py
Comment (JennySmith888) [inline mkidgen3/drivers/ifboard.py:1]: It's likely in the future we will have a new IF board or we might want to support multiple IF boards. For that reason I think it might make more sense to keep the IF board parameters consolidated in their own file. It's few enough parameters that it's not a huge deal if they get mixed in with the FPGA parameters but it will be one more thing to comb through if and when we change IF boards.

File: pyproject.toml
Comment (JennySmith888) [inline pyproject.toml:15]:
```suggestion
authors = [
  {name = "John I. Bailey, III"},
  {name = "Jennifer Pearl Smith"},
  {name = "Aled V. Cuda"},
  {name = "Mazin Lab contributors"}
```

Comment (JennySmith888) [reply pyproject.toml:15]: Should we add ourselves as authors?

**Review by JennySmith888 — DISMISSED** [review]
Looking pretty good so far. Thanks for working on this. Can you add some install instructions in docs or the README? I will test and then approve.

File: pyproject.toml
Comment (ld-cd) [reply pyproject.toml:15]: I think that would make us show up twice on the pypi page, but I'm not sure (https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#authors-maintainers), like I think when its displayed authors is authors union maintainers.

I can check

File: mkidgen3/drivers/ifboard.py
Comment (ld-cd) [reply mkidgen3/drivers/ifboard.py:1]: At present this means that everything in `mkidgen3.server.waveform` can't be imported without installing `pynq` (#14). I think the correct solution here is to move the ifboard driver out of the drivers directory and into some kind non-pl drivers directory along with the anritsu. This would break a number of things though and this patch series is just intended to make things easier to install as opposed to refactoring large portions of code. If you'd like though we can drop this commit and bring it up later when we do a larger refactor.

**Comment by ld-cd** [comment]
Will do

File: pyproject.toml
Comment (ld-cd) [reply pyproject.toml:15]: Ok it shows up like the meta pane here so I think your change makes sense:

https://pypi.org/project/numpy/

**Comment by ld-cd** [comment]
> Looking pretty good so far. Thanks for working on this. Can you add some install instructions in docs or the README? I will test and then approve.

Done

File: mkidgen3/drivers/ifboard.py
Comment (ld-cd) [reply mkidgen3/drivers/ifboard.py:1]: https://github.com/MazinLab/MKIDGen3/commit/237416012bb09f4e0d2f14c57f03c7d7e80c6028 should resolve

File: README.md
Comment (JennySmith888) [inline README.md:14]:
```suggestion
pip install -e '.[client,plotting]'
```

File: pyproject.toml
Comment (JennySmith888) [inline pyproject.toml:66]:
```suggestion
plotting = [
```

**Review by JennySmith888 — COMMENTED** [review]
Added some typo suggestions.

I tested creating a new conda environment on my mac and pip installing and it all worked save for the plotting typo. 
`conda create -n gen3_new python=3.10.4`
`pip install -e '.[client,plotting]'`

I tested creating new virtual environment on the RFSoC and pip installing:
`python3.10 -m venv gen3_new`
`pip install -e '.[server]'`
It errored out with:
`ERROR: Could not find a version that satisfies the requirement xrfdc==1.0; extra == "board" (from mkidgen3[board,sccommon]) (from versions: none)
ERROR: No matching distribution found for xrfdc==1.0; extra == "board"`
I'm not sure if that's expected. I see a note in the `.toml` about `xrfdc` being a problem.

**Comment by ld-cd** [comment]
Will add a note about that in the README.md, highly tempted to just package xrfdc and xrfclk on pypi

**Comment by ld-cd** [comment]
(Don't merge this yet, I'm gonna do some rebasing so the history looks sane)

**Pull request merged**
